src/pspm_dcm.m

```diff
@@ -582,6 +583,26 @@
     return;
 end
 
+% Remove the trial that has insufficient information
+sbs_data_out = sbs_data;
+flag_valid = ~cellfun(@isempty, sbs_trlstart);
+error_log = zeros(size(sbs_iti));
+% Do processing in the index of valid sessions
+idx_session = nonzeros((1:size(sbs_data,1)).*flag_valid);
+for i_session = idx_session'
+    % Find the position of the target trial in proc_subsessions
+    % Check the interval since the start of the last trial
+    error_log(i_session)=sum(sbs_iti{i_session}<options.lasttrl);
+    if error_log(i_session) > 0
+        i_trial = length(sbs_iti{i_session});
+        last_trl_start = sbs_trlstart{i_session};
+        last_trl_start = last_trl_start(i_trial)*model.sr;
+        last_trl_stop = numel(sbs_data{i_session,1});
+        sbs_data_out{i_session,1}(last_trl_start:last_trl_stop) = [];
+    end

```

Comment:
In other words: why are you not using in this "if" statement the information about which ITI is too short ?

Reply:
Sorry I am not quite clear about this question. Instead, I can explain the idea here. The idea is we try to find the sessions where the last trial is with insufficient information and remove the last trial in such sessions. Such sessions with this issue will be recorded into the variable _error_log_.

Reply:
Do you store this variable in the final model structure ?

Reply:
Yes, I have updated the code to add storing this variable in the final model structure as _dcm.lasttrialfilter_.